homeassistant/components/eheimdigital/light.py

```diff
@@ -0,0 +1,115 @@
+"""EHEIM Digital lights."""
+
+from typing import Any
+
+from eheimdigital.classic_led_ctrl import EheimDigitalClassicLEDControl
+from eheimdigital.types import LightMode
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_EFFECT,
+    EFFECT_OFF,
+    ColorMode,
+    LightEntity,
+    LightEntityFeature,
+)
+from homeassistant.const import Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.util.color import brightness_to_value, value_to_brightness
+
+from . import EheimDigitalConfigEntry
+from .const import EFFECT_DAYCL_MODE, EFFECT_TO_LIGHT_MODE
+from .coordinator import EheimDigitalUpdateCoordinator
+from .entity import EheimDigitalEntity
+
+BRIGHTNESS_SCALE = (1, 100)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: EheimDigitalConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the callbacks for the coordinator so lights can be added as devices are found."""
+    coordinator = entry.runtime_data
+
+    async def async_setup_device_entities(device_address: str) -> None:
+        """Set up the light entities for a device."""
+        device = coordinator.hub.devices[device_address]
+        entities: list[EheimDigitalClassicLEDControlLight] = []
+
+        if isinstance(device, EheimDigitalClassicLEDControl):
+            for channel in range(2):
+                if len(device.tankconfig[channel]) > 0:
+                    entities.append(
+                        EheimDigitalClassicLEDControlLight(coordinator, device, channel)
+                    )
+                    coordinator.known_devices.add(device.mac_address)
+        async_add_entities(entities)
+
+    coordinator.add_platform_callback(Platform.LIGHT, async_setup_device_entities)
```

Comment:
Don't forget that this leads to that devices will only be loaded after the second coordinator update, since the coordinator already is updated. So I would suggest calling the function after registering it as well